### PR TITLE
groff: add page

### PR DESCRIPTION
**Pull request by waldyrious**
Sources: https://embswit.wordpress.com/2013/02/08/render-man-page-from-source/, and the EXAMPLES section of `man groff` :)

----

- [x] The page (if new), does not already exist in the repo.

- [x] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: pages/common/groff.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+`groff -man -T utf8 {{manpage.1}}`
+
+- Render a man page using the ASCII output device, and display it using a pager:
+
+`groff -man -T ascii {{manpage.1}} | less`
+
+- Render a man page into an HTML file:
+
+`groff -man -T html {{manpage.1}} > page.html`
```
Comment (agnivade) [inline pages/common/groff.md]: tokens around page.html

File: pages/common/groff.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# groff
+
+> Typesetting program that reads plain text mixed with formatting commands and produces formatted output.
+> It is the GNU replacement for the troff and nroff Unix commands for text formatting.
```
Comment (agnivade) [inline pages/common/groff.md]: Should we wrap troff and nroff with backticks ?

File: pages/common/groff.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+`groff -man -T html {{manpage.1}} > page.html`
+
+- Process a roff file using the `tbl` and `pic` preprocessors, and the `me` macro set:
+
+`groff -t -p -me -T utf8 {{foo.me}}`
+
+- Run a `groff` command with preprocessor and macro options guessed by the `grog` utility:
+
+`eval "$(grog -T utf8 {{foo.me}})"`
```
Comment (agnivade) [inline pages/common/groff.md]: Hmm .. this seems a bit arcane. Does the `grog` command itself output the `groff` command ?

Comment (waldyrious) [reply pages/common/groff.md]: Yeah. This example is actually [mentioned in the groff manpage](http://man.gnu.org.ua/manpage/?1+groff#EXAMPLES).

File: pages/common/groff.md (hunk 2 above)
Comment (waldyrious) [reply pages/common/groff.md]: Yes, good catch.

**Comment by waldyrious** [comment]
By the way, I noticed there's additional example-based documentation in the manual [current version](http://www.hep.by/gnu/groff/Invocation-Examples.html#fnd-1), [old version](http://web.cecs.pdx.edu/~trent/gnu/groff/groff.html#SEC13).

Do you see anything there that could be added here, or even replace / improve some of the examples in this page?

File: pages/common/groff.md (hunk 3 above)
Comment (sbrl) [reply pages/common/groff.md]: That's weird! Looks ok then to me. Either that or something like `grog command | bash` I guess. I can see how `eval` is more explicit though - I vote for eval then.

File: pages/common/groff.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# groff
+
+> Typesetting program that reads plain text mixed with formatting commands and produces formatted output.
+> It is the GNU replacement for the `troff` and `nroff` Unix commands for text formatting.
+
+- Render a man page as plain text, and display the result:
+
+`groff -man -T utf8 {{manpage.1}}`
+
+- Render a man page using the ASCII output device, and display it using a pager:
```
Comment (sbrl) [inline pages/common/groff.md]: `device`? That's a bit strange. Could we not use `plugin` or `module` here? I for one associate `device` with a physical device or chip on my motherboard.

Comment (waldyrious) [reply pages/common/groff.md]: "output device" is the term they use in the [documentation](http://man.gnu.org.ua/manpage/?1+groff#OPTIONS):

```
−T dev: Set output device to dev.

  Real devices in groff are:

  - dvi     TeX DVI format.
  - html
  - xhtml   HTML and XHTML output.
  - lbp     Canon CAPSL printers (LBP-4 and LBP-8 series laser printers).
  - lj4     HP LaserJet4 compatible (or other PCL5 compatible) printers.
  - ps      PostScript output.

  For the following TTY output devices, −T selects the output encoding:
	
  - ascii   7bit ASCII.
  - cp1047  Latin-1 character set for EBCDIC hosts.
  - latin1  ISO 8859-1.
  - utf8    Unicode character set in UTF-8 encoding.

  (...)

  The default device is ps.
```

Comment (sbrl) [reply pages/common/groff.md]: Ah, ok! That's really weird.

Comment (waldyrious) [reply pages/common/groff.md]: Yeah... I guess when some of these older stuff were created in a time when naming conventions weren't as established as they are today (e.g. maybe "device" wasn't as strongly correlated to physical devices as it is today)

**Pull request merged**